File: practica2_820574_839304/java/Main.java
/*
* Fichero: Main.java
* Autores: Alicia Lazaro Huerta y Manel Jorda Puig Rubio
* Fecha: 27/02/2023
*/

// Clase principal
// Lista y ejecuta los 3 programas implementados, uno seguido de otro
public class Main {
    public static void main(String[] args) {
        
        Prog_suma p1 = new Prog_suma();
        Prog_cuenta_atras p2 = new Prog_cuenta_atras();
        Prog_factorial p3 = new Prog_factorial();

        p1.listar();
        p1.run();
        System.out.println("\n");

        p2.listar();
        p2.run();
        System.out.println("\n");


        p3.listar();
        p3.run();
    }
}

File: practica2_820574_839304/c++/main.cc
/*
* Fichero: main.cc
* Autores: Alicia Lazaro Huerta y Manel Jorda Puig Rubio
* Fecha: 27/02/2023
*/

#include "prog_suma.h"
#include "prog_cuenta_atras.h"
#include "prog_factorial.h"
#include <iostream>

// Programa principal
// Lista y ejecuta los 3 programas implementados, uno seguido de otro
int main(){
    Prog_Suma suma_exe;
    suma_exe.listar();
    suma_exe.run();
    
    cout <<endl;

    Prog_Cuenta_Atras cuenta_atras_exe;
    cuenta_atras_exe.listar();
    cuenta_atras_exe.run();

    cout << endl;
    
    Prog_Factorial factorial_exe;
    factorial_exe.listar();
    factorial_exe.run();

    return 0;
}
